align/schema/constraint.py

```diff
@@ -596,6 +596,26 @@ def check(self, checker):
         checker.append(checker.cast(bvar.urx-bvar.llx, float) < self.ratio_high*checker.cast(bvar.ury-bvar.lly, float))
 
 
+class Boundary(HardConstraint):
+    """
+    Define `max_height` and/or `max_width` on a subcircuit in micrometers.
+    """
+    subcircuit: str
+    max_width: Optional[float] = 10000
+    max_height: Optional[float] = 10000
+
+    def check(self, checker):
+        bvar = checker.bbox_vars(self.subcircuit, is_subcircuit=True)
+
+        if self.max_width is not None:
+            assert self.max_width >= 0, f'Boundary:max_width should be greater than zero {self.max_width}'
+            checker.append(checker.cast(bvar.urx-bvar.llx, float) <= 1000*self.max_width) # in nanometer
```

Comment:
I think you need to convert to layers.json units here since that should be the units inside the checker.
So that should be 1000*scale_factor, not just 1000. Plus there might be issues with floating point rounding here.
So maybe we need to convert 1000*self.max_width to an int (and make sure there is not a large error.) I guess, what units do we want to use for this?

Reply:
This implementation is using nm as the intermedia unit between constraint and checker as I chose not to load PDK content to constraint.py